makefiles/firmware.mk

```diff
@@ -101,7 +101,7 @@
 # If PX4_BASE wasn't set previously, work out what it should be
 # and set it here now.
 #
-MK_DIR			?= $(dir $(lastword $(MAKEFILE_LIST)))
+MK_DIR			:= $(dir $(lastword $(MAKEFILE_LIST)))
```

Comment:
It doesn't behave correctly for makefiles in target subdirs
(Firmware/makefiles/nuttx) when the value is not set immediately.

On Mon, Jun 15, 2015 at 11:37 PM, Lorenz Meier <notifications@github.com>
wrote:

> In makefiles/firmware.mk
> <https://github.com/PX4/Firmware/pull/2388#discussion_r32494409>:
>
> > @@ -101,7 +101,7 @@
> >  # If PX4_BASE wasn't set previously, work out what it should be
> >  # and set it here now.
> >  #
> > -MK_DIR			?= $(dir $(lastword $(MAKEFILE_LIST)))
> > +MK_DIR			:= $(dir $(lastword $(MAKEFILE_LIST)))
>
> I think APM uses the override in its build, so this needs to stay ?=
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PX4/Firmware/pull/2388/files#r32494409>.
>